### PR TITLE
Extensions backward compatibility

### LINKED ISSUE #26: Error logged in Python output window -Error: command 'completion/itemSelected' already exists
## Environment data

- VS Code version: Insiders
- Extension version (available under the Extensions sidebar): Dev
- OS and version: Windows 10
- Python version (& distribution if applicable, e.g. Anaconda): N/A
- Type of virtual environment used (N/A | venv | virtualenv | conda | ...): N/A
- Relevant/affected Python packages and their versions: N/A

## Actual behavior

Error in Python Extension - Error: command 'completion/itemSelected' already exists

## Expected  behavior

No Errors.

## Steps to reproduce:
1. Open VSC
2. Open a Python file

## Logs
Output for `Python` in the `Output` panel (`View`→`Output`, change the drop-down the upper-right of the `Output` panel to `Python`)

```
Starting Microsoft Python language server.
[Error - 1:46:13 PM] Server initialization failed.
Error: command 'completion/itemSelected' already exists
	at e.registerCommand (c:\Program Files\Microsoft VS Code Insiders\resources\app\out\vs\workbench\node\extensionHostProcess.js:689:906)
	at Object.registerCommand (c:\Program Files\Microsoft VS Code Insiders\resources\app\out\vs\workbench\node\extensionHostProcess.js:762:949)
	at ExecuteCommandFeature.register (C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\node_modules\vscode-languageclient\lib\client.js:1501:53)
	at ExecuteCommandFeature.initialize (C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\node_modules\vscode-languageclient\lib\client.js:1491:14)
	at LanguageClient.initializeFeatures (C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\node_modules\vscode-languageclient\lib\client.js:2168:21)
	at connection.initialize.then (C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\node_modules\vscode-languageclient\lib\client.js:1888:18)
	at <anonymous>
Initializing for C:\ProgramData\Anaconda3\python.exe
Initializing for C:\ProgramData\Anaconda3\python.exe
[Error - 1:46:20 PM] System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.PythonTools.Interpreter.Ast.AstPythonInterpreter.ImportModule(String name)
   at Microsoft.PythonTools.Analysis.ModuleTable.TryImport(String name, ModuleReference& res)
   at Microsoft.PythonTools.Analysis.Analyzer.DDG.TryImportModule(String modName, Boolean forceAbsolute, ModuleReference& moduleRef, IReadOnlyList`1& remainingParts)
   at Microsoft.PythonTools.Analysis.Analyzer.DDG.Walk(FromImportStatement node)
   at Microsoft.PythonTools.Parsing.Ast.FromImportStatement.Walk(PythonWalker walker)
   at Microsoft.PythonTools.Analysis.Analyzer.DDG.Walk(SuiteStatement node)
   at Microsoft.PythonTools.Parsing.Ast.SuiteStatement.Walk(PythonWalker walker)
   at Microsoft.PythonTools.Parsing.Ast.PythonAst.Walk(PythonWalker walker)
   at Microsoft.PythonTools.Analysis.AnalysisUnit.AnalyzeWorker(DDG ddg, CancellationToken cancel)
   at Microsoft.PythonTools.Analysis.Analyzer.DDG.Analyze(Deque`1 queue, CancellationToken cancel, Action`1 reportQueueSize, Int32 reportQueueInterval)
   at Microsoft.PythonTools.Analysis.PythonAnalyzer.AnalyzeQueuedEntries(CancellationToken cancel)
   at Microsoft.PythonTools.Intellisense.AnalysisQueue.GroupAnalysis.Analyze(CancellationToken cancel)
   at Microsoft.PythonTools.Intellisense.AnalysisQueue.HandleAnalyzable(IAnalyzable item, AnalysisPriority priority, CancellationToken cancellationToken)
   at Microsoft.PythonTools.Intellisense.AnalysisQueue.ConsumerLoop()
Reloading modules...
Reloading modules...

Unhandled Exception: System.ObjectDisposedException: PriorityProducerConsumer`1 instance is disposed
Object name: 'PriorityProducerConsumer`1'.
   at Microsoft.PythonTools.Analysis.Infrastructure.DisposeToken.ThrowIfDisposed()
   at Microsoft.PythonTools.Analysis.Infrastructure.PriorityProducerConsumer`1.Produce(T value, Int32 priority)
   at Microsoft.PythonTools.Intellisense.AnalysisQueue.Enqueue(IAnalyzable item, AnalysisPriority priority)
   at Microsoft.PythonTools.Analysis.LanguageServer.Server.ReloadModulesAsync(CancellationToken token)
   at Microsoft.PythonTools.Analysis.Infrastructure.TaskExtensions.DoNotWait(Task task)
   at Microsoft.Python.LanguageServer.Implementation.LanguageServer.<DidChangeConfiguration>b__29_0()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
[Info  - 1:46:22 PM] Connection to server got closed. Server will restart.
[Error - 1:46:22 PM] Request textDocument/documentSymbol failed.
Error: Connection got disposed.
	at Object.dispose (C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\node_modules\vscode-jsonrpc\lib\main.js:825:25)
	at Object.dispose (C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\node_modules\vscode-languageclient\lib\client.js:57:35)
	at LanguageClient.handleConnectionClosed (C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\node_modules\vscode-languageclient\lib\client.js:2036:42)
	at LanguageClient.handleConnectionClosed (C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\node_modules\vscode-languageclient\lib\main.js:127:15)
	at closeHandler (C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\node_modules\vscode-languageclient\lib\client.js:2023:18)
	at CallbackList.invoke (C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\node_modules\vscode-jsonrpc\lib\events.js:71:39)
	at Emitter.fire (C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\node_modules\vscode-jsonrpc\lib\events.js:135:36)
	at closeHandler (C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\node_modules\vscode-jsonrpc\lib\main.js:221:26)
	at CallbackList.invoke (C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\node_modules\vscode-jsonrpc\lib\events.js:71:39)
	at Emitter.fire (C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\node_modules\vscode-jsonrpc\lib\events.js:135:36)
	at StreamMessageReader.AbstractMessageReader.fireClose (C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\node_modules\vscode-jsonrpc\lib\messageReader.js:135:27)
	at Socket.<anonymous> (C:\Users\dojayama.REDMOND\.vscode-insiders\extensions\pythonVSCode\node_modules\vscode-jsonrpc\lib\messageReader.js:188:62)
	at emitOne (events.js:101:20)
	at Socket.emit (events.js:191:7)
	at Pipe._handle.close [as _onclose] (net.js:510:12)
Initializing for C:\ProgramData\Anaconda3\python.exe
Reloading modules...
Reloading modules...
Reloading modules...
```   

Output from `Console` under the `Developer Tools` panel (toggle Developer Tools on under `Help`)

```
XXX
```

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
- Declare old IServer
- Implement IServer stub 
- Implement thunk down to the old type of completion handling

Also fixed #26 by removing the command. IC extension is registering its own command, we do not handle this one anyway.

### CODE REVIEW

File: src/Analysis/Engine/Impl/LanguageServer/Extensibility.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            Tree = tree;
+            Location = location;
+            CompletionList = initialCompletionList;
+        }
+
+        public ModuleAnalysis Analysis { get; }
+        public PythonAst Tree { get; }
+        public SourceLocation Location { get; }
+
+        public CompletionList CompletionList;
```
Comment (AlexanderSher) [inline src/Analysis/Engine/Impl/LanguageServer/Extensibility.cs]: Should it be a property?

File: src/Analysis/Engine/Impl/LanguageServer/Server.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+// permissions and limitations under the License.
+
+using System;
+using Microsoft.PythonTools.Analysis.Infrastructure;
+using Microsoft.PythonTools.Parsing.Ast;
+
+namespace Microsoft.PythonTools.Analysis.LanguageServer {
+    [Obsolete("Implement Microsoft.Python.LanguageServer.Extensions.ILanguageServerExtension model")]
+    public class Server : IServer {
+        public event EventHandler<Extensibility.CompletionEventArgs> PostProcessCompletion;
```
Comment (AlexanderSher) [inline src/Analysis/Engine/Impl/LanguageServer/Server.cs]: This isn't part of the interface. Is that correct?

File: src/Analysis/Engine/Impl/LanguageServer/Extensibility.cs (hunk 1 above)
Comment (MikhailArkhipov) [reply src/Analysis/Engine/Impl/LanguageServer/Extensibility.cs]: It the new world, sure. This is old structs copied verbatim from previous incarnation.

File: src/Analysis/Engine/Impl/LanguageServer/Server.cs (hunk 2 above)
Comment (MikhailArkhipov) [reply src/Analysis/Engine/Impl/LanguageServer/Server.cs]: Yep, matches old case and this is why IC ended up casting.

**Pull request merged**